src/callbacks.rs

```diff
@@ -36,61 +43,62 @@ impl SelfCallbacks for Contract {
                 if is_final {
                     remove_indices.push(index);
                 }
-                total_balance += claim_amount.0;
+                total_balance = total_balance.saturating_add(claim_amount);
                 let event = FtLockupClaimLockup {
                     id: index,
                     amount: claim_amount,
                 };
                 events.push(event);
             }
             if !remove_indices.is_empty() {
-                let mut indices = self.account_lockups.get(&account_id).unwrap_or_default();
+                let mut indices = self
+                    .account_lockups
+                    .get(&account_id)
+                    .unwrap_or(UnorderedSet::new(StorageKey::AccountLockups));
                 for index in remove_indices {
                     indices.remove(&index);
                 }
                 self.internal_save_account_lockups(&account_id, indices);
             }
-            emit(EventKind::FtLockupClaimLockup(events));
+            // TODO: Should we emit a single vector or all separate?
+            events.iter().for_each(|event| event.emit());
```

Comment:
Firing individual logs vs batching is a matter of gas usage. If I remember correctly from other testing, we can fire up to 300 events with 300TGas (max limit).

How many events do we expect here? Why not batching?